crates/router/src/core/webhooks/outgoing_v2.rs

```diff
@@ -0,0 +1,817 @@
+                type_name!(domain::Event),
+                CryptoOperation::Encrypt(
+                    response_to_store
+                        .encode_to_string_of_json()
+                        .change_context(
+                            errors::WebhooksFlowError::OutgoingWebhookResponseEncodingFailed,
+                        )
+                        .map(Secret::new)?,
+                ),
+                Identifier::Merchant(merchant_key_store.merchant_id.clone()),
+                merchant_key_store.key.get_inner().peek(),
+            )
+            .await
+            .and_then(|val| val.try_into_operation())
+            .change_context(errors::WebhooksFlowError::WebhookEventUpdationFailed)
+            .attach_printable("Failed to encrypt outgoing webhook response content")?,
+        ),
+    };
+    state
+        .store
+        .update_event_by_merchant_id_event_id(
+            key_manager_state,
+            merchant_id,
+            event_id,
+            event_update,
+            &merchant_key_store,
+        )
+        .await
+        .change_context(errors::WebhooksFlowError::WebhookEventUpdationFailed)
+}
+
+fn increment_webhook_outgoing_received_count(merchant_id: &common_utils::id_type::MerchantId) {
+    metrics::WEBHOOK_OUTGOING_RECEIVED_COUNT.add(
+        &metrics::CONTEXT,
+        1,
+        &[metrics::KeyValue::new(
+            MERCHANT_ID,
+            merchant_id.get_string_repr().to_owned(),
+        )],
+    )
+}
+
+async fn success_response_handler(
+    state: SessionState,
+    merchant_id: &common_utils::id_type::MerchantId,
+    process_tracker: Option<storage::ProcessTracker>,
+    business_status: &'static str,
+) -> CustomResult<(), errors::WebhooksFlowError> {
+    increment_webhook_outgoing_received_count(merchant_id);
+
+    match process_tracker {
+        Some(process_tracker) => state
+            .store
+            .as_scheduler()
+            .finish_process_with_business_status(process_tracker, business_status)
+            .await
+            .change_context(
+                errors::WebhooksFlowError::OutgoingWebhookProcessTrackerTaskUpdateFailed,
+            ),
+        None => Ok(()),
+    }
+}
+
+async fn error_response_handler(
+    _state: SessionState,
+    merchant_id: &common_utils::id_type::MerchantId,
+    delivery_attempt: enums::WebhookDeliveryAttempt,
+    status_code: u16,
+    log_message: &'static str,
+    _schedule_webhook_retry: ScheduleWebhookRetry,
+) -> CustomResult<(), errors::WebhooksFlowError> {
+    metrics::WEBHOOK_OUTGOING_NOT_RECEIVED_COUNT.add(
+        &metrics::CONTEXT,
+        1,
+        &[metrics::KeyValue::new(
+            MERCHANT_ID,
+            merchant_id.get_string_repr().to_owned(),
+        )],
+    );
+
+    let error = report!(errors::WebhooksFlowError::NotReceivedByMerchant);
+    logger::warn!(?error, ?delivery_attempt, status_code, %log_message);
+
+    //TODO: add outgoing webhook retries support
```

Comment:
why still these are todo?

Reply:
Currently outgoing webhook retries are not supported as part of this PR, that's why these TODOs